packages/utils/src/baggage.ts

```diff
@@ -179,8 +179,13 @@ export function parseBaggageSetMutability(
   sentryTraceHeader: TraceparentData | string | false | undefined | null,
 ): Baggage {
   const baggage = parseBaggageHeader(rawBaggageValue || '');
-  if (!isSentryBaggageEmpty(baggage) || (sentryTraceHeader && isSentryBaggageEmpty(baggage))) {
-    setBaggageImmutable(baggage);
-  }
+
+  // Because we are always creating a Baggage object by calling `parseBaggageHeader` above
+  // (either a filled one or an empty one, even if we didn't get a `baggage` header),
+  // we only need to check if we have a sentry-trace header or not. As soon as we have it,
+  // we set baggage immutable. In case we don't get a sentry-trace header, we can assume that
+  // this SDK is the head of the trace and thus we still permit mutation at this time.
+  sentryTraceHeader && setBaggageImmutable(baggage);
+
```

Comment:
This is arguably a critical change here because it does not cover one edge case (which I don't know if it ever exists): We receive an incoming request **without** a `sentry-trace` header but **with** a `baggage` header that contains `sentry-` entries. In such a situation, we would keep the baggage mutable. 
WDYT, should we handle this edge case or do we assume that we won't receive this combination of headers?

I personally can't think of a situation when this would occur because all SDKs already send `sentry-trace` headers. I therefore made this simplification here. Happy to revisit though!

The inverse situation (no baggage but a sentry-trace header) is of course handled correctly.

Reply:
> but I think we should just in case
@lforst and I also just discussed this again and I just handled it in [795dd25](https://github.com/getsentry/sentry-javascript/pull/5319/commits/795dd25bb5ec16d81c10c4d3afeb642aba1a688c).

Reply:
> Let's update the develop docs to account for this

Sure, I'm all for it. We should try to find a conclusion as to what this means though. If we get a baggage header with sentry- entries but no sentry-trace header,
* Are we the head of trace or not?
* What happened before (i.e. in the SDK that propagated the baggage header to us) and where could something like this come from?
* And most importantly, is it okay to say, we just freeze baggage and propagate it downstream? 
* Would we in this case get `sentry-trace_id` with that baggage header and if yes, why don't we get sentry-trace in that situation? 
* Could this come from a third party that (intentionally) sends sentry- entries in baggage s.t. our DS data is corrupted?